### ISSUE TITLE
Issue with implementation

### ISSUE THREAD

**Issue by ashrithaMula** [tmahlburg/picosoc-basys3#4]
I am trying to run implementation in vivado but getting a place error and also insufficient I/O ports. I am also getting IO clock placer failed.

**Comment by tmahlburg** [comment]
Do you have this file in your project?
https://github.com/Digilent/Basys3/blob/master/Resources/XDC/Basys3_Master.xdc
You need to uncomment the pins and the clock used here in that file.

**Comment by ashrithaMula** [comment]
Hello!
Yes, I did. But, still getting the same placing error that the resources
are not sufficient.(IO ports).

**Comment by tmahlburg** [comment]
Did you follow this tutorial by digilent to set up vivado?
https://digilent.com/reference/programmable-logic/guides/getting-started-with-vivado
Especially the "Adding the constraints file"?
And is `basys3.v` set as the top file of the implementation?

**Comment by tmahlburg** [comment]
Uncommented should be lines 7-9, 47-78, 267-270 and 285-294, btw.

**Comment by ashrithaMula** [comment]
Yes I did. Can you share me the picosoc code. Did you change anything in
that?

**Comment by tmahlburg** [comment]
No, I didn't change anything in there. Even if I did, it would not explain the placement error. Can you share your .xdc file and the exact error message?

**Comment by ashrithaMula** [comment]
Hello. I sent you my constraint file. Can you please check it once.

**Comment by ashrithaMula** [comment]
Hii. That over utilization issue is resolved. I just had one doubt. Where
did you store the instructions? How can we use flash chip to store the
instructions?

**Comment by tmahlburg** [comment]
Glad to hear that.
You can just load your compiled .bin file, this forum post describes how it should work:
https://forum.digilent.com/topic/227-how-to-program-the-quad-spi-flash-on-the-basys3/

**Comment by tmahlburg** [comment]
You can also look at my other project [picoramsoc](https://github.com/tmahlburg/picoramsoc), if you want to load the instructions directly into the RAM.
